### ISSUE TITLE
Alternate Onion Skin For Drawing Instead of Frame

### ISSUE THREAD

**Issue by MitchWoodin** [opentoonz/opentoonz#4659]
### Is your feature request related to a problem? Please describe

I feel as though I can't truly work while inbetweening in opentoonz because the onion skin doesn't automatically show previous and next drawings. Rather it shows the previous and next frames. When different drawing are not spaced evenly apart this means that I am having to manually set the onion skin to the correct drawings rather than having the program do this.

### Describe the solution you'd like

I believe there should be an alternate onion skin option available to have a user defined set of previous/next drawings instead of frames. That way when drawing are spaced unevenly they will be correctly visible in the onion skin without having to manually update the needed frames.

**Comment by artisteacher** [comment]
Have you tried using onion skin with the level strip active?

**Comment by MitchWoodin** [comment]
The issue I have with that is that the bg changes from white to grey. Is there a way to set the level bg to also be white?

**Comment by MitchWoodin** [comment]
The other issue is if I need to reorder my drawings in the timeline that isn't reflected in the level strip.

**Comment by artisteacher** [comment]
> Is there a way to set the level bg to also be white?

Levels are transparent, but you can change the level editor box color in Preferences, in the Color tab. 

> The other issue is if I need to reorder my drawings in the timeline that isn't reflected in the level strip.

There's a preference that should be enabled by default (under Xsheet?) that should keep the level strip order synced with the timeline. Though you may need to use the auto-renumber command in the timeline's context menu if you want the level strip order to change based upon reordering in the Timeline.

**Comment by MitchWoodin** [comment]
It seems as though the level strip doesn't update when I reorder drawings in the Xsheet. It will renumber them when I create a new drawing. But if I drag around the drawings in the Xsheet it doesn't update

**Comment by artisteacher** [comment]
Just use the auto-renumber command when you are done reordering:
<img width="200" alt="Screen Shot 2022-12-19 at 7 02 16 PM" src="https://user-images.githubusercontent.com/21159116/208551186-d538a9a9-d6e3-47fb-a196-6374935bee22.png">
There's also a renumber command in the level strip if you need it on that end of things.

**Comment by MitchWoodin** [comment]
This definitely works but it's unintuitive and from a practical stand point it's still very inefficient. It still makes a lot of sense to implement the feature I requested initially. The solution you've outlined is a band-aid solution to the problem I believe. New users aren't going to know to follow all those steps just for basic onion skins to work.

**Comment by artisteacher** [comment]
Yes, it's definitely a workaround and with the very specific way you want to work, not the most efficient or intuitive. 

To be honest, I don't understand why the numbers staying synced is quite so important. The order of drawings in the level strip is often not that meaningful since drawings can be exposed and reused in any configuration in the Timeline and they might get reused in different configurations in other scenes.

In my opinion the real failing of the level strip as a workaround, is that you can't see the other levels. Ie. if you need a character to interact with the environment, you can't see the character and environment at the same time in the level strip unless they are on the same level.

We do have working basic onionskin already, without any workarounds. I think for the average new user, that's all they need. It works great when using tweening, cut-out , and mesh deformation techniques. It also works great when drawing if your drawings are evenly spaced.

That's not to say that I don't see value in the request - I think it could be quite helpful and speed things up when using onionskin specifically for new drawings that aren't spaced evenly.

**Comment by RodneyBaker** [comment]
I haven't read all of the discussion but...
If the option to have the xsheet and level strip stay in sync is turned on in Preferences does each drawing being the same as a frame make any difference?  There won't be any missing drawings with this approach.

**Comment by MitchWoodin** [comment]
The drawing don't stay in the correct order using that option. The reason they need to stay in order for this use case is so when you're inbetweening you're seeing the correct drawings either side of the one you're currently drawing. There won't be missing drawings but you won't be seeing the correct ones while inbetweening.

I also agree that not being able to see the other layers is another reason this isn't a great approach. It isn't needing the numbers to be in sync the need is for the drawings to be in the same order in the xsheet and the level strip.

**Comment by konero** [comment]
Use Shift & Trace for inbetweening.

You can use the F1, F2 and F3 hotkeys to show/hide the drawings.

F1: Show previous reference drawing
F2: Hide previous and next reference drawings
F3: Show next reference drawing

You can also shift the reference drawings if you need to trace some parts, give it a try.

![image](https://user-images.githubusercontent.com/19820721/208666594-17c9287c-6fb7-4c8f-9196-4db3c364e0bb.png)

**Comment by MitchWoodin** [comment]
This is a much better solution. However I don't seem to have those hotkeys. Is there a hotkey to reset the shift markers?

**Comment by MitchWoodin** [comment]
Never mind about the F key hotkey I do have them. They weren't working intially for some reason
